### PR TITLE
PIMS-30: Issue with Fiscal Year reporting on excel extract

### PR DESCRIPTION
**Pull request by LawrenceLau2020**

### CODE REVIEW

File: backend/api/Areas/Reports/Models/Project/ProjectModel.cs [hunk 1]
```diff
@@ -27,12 +27,12 @@ public class ProjectModel
         /// <summary>
         /// get/set - The reported fiscal year this project.
         /// </summary>
-        public int ReportedFiscalYear { get; set; }
+        public string ReportedFiscalYear { get; set; }
```
Comment (Fosol) [inline backend/api/Areas/Reports/Models/Project/ProjectModel.cs:30]: Any reason for the string requirement?  Isn't it just an `int`?

**Review by Fosol — APPROVED** [review]

File: backend/api/Areas/Reports/Models/Project/ProjectModel.cs (hunk 1 above)
Comment (LawrenceLau2020) [reply backend/api/Areas/Reports/Models/Project/ProjectModel.cs:30]: The output for the reported fiscal year is actually a string: "21/22"

Comment (LawrenceLau2020) [reply backend/api/Areas/Reports/Models/Project/ProjectModel.cs:30]: Ok, it looks like my change has affected some of the backend tests....

**Review by jdtoombs — APPROVED** [review]

**Comment by LawrenceLau2020** [comment]
Please review my changes when you have a chance. Because the project model stores the fiscal year as an integer, I created a string field for the ReportedFiscalYear and the ActualFiscalYear.

I also had to update the FiscalYear method in the DateExtensions class, to handle cases where the fiscalYear was being passed in as 0 in some test cases.

File: backend/core/Extensions/DateExtensions.cs [hunk 2]
```diff
@@ -33,7 +33,7 @@ public static int GetFiscalYear(this DateTime date)
         /// <returns></returns>
         public static string FiscalYear(this int fiscalYear)
         {
-            return $"{(fiscalYear - 1).ToString().Substring(2, 2)}/{(fiscalYear).ToString().Substring(2, 2)}";
+            return fiscalYear > 0 ? $"{(fiscalYear - 1).ToString().Substring(2, 2)}/{(fiscalYear).ToString().Substring(2, 2)}" : "";
```
Comment (LawrenceLau2020) [inline backend/core/Extensions/DateExtensions.cs:36]: During debugging, it seems that one of the tests which instantiates a new project, passes in the fiscalYear as 0 which caused this to fail.

File: backend/api/Areas/Reports/Mapping/Project/ProjectMap.cs [hunk 3]
```diff
@@ -34,6 +35,8 @@ public void Register(TypeAdapterConfig config)
                 .Map(dest => dest.Description, src => src.Description)
                 .Map(dest => dest.ReportedFiscalYear, src => src.ReportedFiscalYear)
                 .Map(dest => dest.ActualFiscalYear, src => src.ActualFiscalYear)
+                .Map(dest => dest.ReportedFiscalYearString, src => src.ReportedFiscalYear.FiscalYear())
```
Comment (LawrenceLau2020) [inline backend/api/Areas/Reports/Mapping/Project/ProjectMap.cs:38]: I needed to add 2 new fields to the mapping, so now the report will show both the fiscal year as well as the formatted fiscal year "string" for both the reported and actual fiscal year.

File: backend/core/Extensions/DateExtensions.cs (hunk 2 above)
Comment (Fosol) [reply backend/core/Extensions/DateExtensions.cs:36]: You probably could have just updated the test to set the default value of "20/21".

**Review by Fosol — APPROVED** [review]

**Pull request merged**